index.js

```diff
@@ -98,89 +97,81 @@ module.exports = function(opts = {}) {
-          const workerParametersStartIndex = matchIndex + partBeforeArgs.length;
-
-          let workerIdPromise;
-          if (workerSource === "import.meta.url") {
-            // Turn the current file into a chunk
-            workerIdPromise = Promise.resolve(id);
-          } else {
-            // Otherwise it's a string literal either directly or in the `new URL(...)`.
-            if (directWorkerFile) {
-              const fullMatchWithOpts = `${fullMatch}, …)`;
-              const fullReplacement = `new Worker(new URL(${directWorkerFile}, import.meta.url), …)`;
-
-              if (!longWarningAlreadyShown) {
-                this.warn(
-                  `rollup-plugin-off-main-thread:
+        ] = match;
+
+        const workerParametersEndIndex = match.index + fullMatch.length;
+        const matchIndex = match.index;
+        const workerParametersStartIndex = matchIndex + partBeforeArgs.length;
+
+        let workerIdPromise;
+        if (workerSource === "import.meta.url") {
+          // Turn the current file into a chunk
+          workerIdPromise = Promise.resolve(id);
+        } else {
+          // Otherwise it's a string literal either directly or in the `new URL(...)`.
+          if (directWorkerFile) {
+            const fullMatchWithOpts = `${fullMatch}, …)`;
+            const fullReplacement = `new Worker(new URL(${directWorkerFile}, import.meta.url), …)`;
+
+            if (!longWarningAlreadyShown) {
+              this.warn(
+                `rollup-plugin-off-main-thread:
 \`${fullMatchWithOpts}\` suggests that the Worker should be relative to the document, not the script.
 In the bundler, we don't know what the final document's URL will be, and instead assume it's a URL relative to the current module.
 This might lead to incorrect behaviour during runtime.
 If you did mean to use a URL relative to the current module, please change your code to the following form:
 \`${fullReplacement}\`
 This will become a hard error in the future.`,
-                  matchIndex
-                );
-                longWarningAlreadyShown = true;
-              } else {
-                this.warn(
-                  `rollup-plugin-off-main-thread: Treating \`${fullMatchWithOpts}\` as \`${fullReplacement}\``,
-                  matchIndex
-                );
-              }
-              workerFile = directWorkerFile;
-            }
-
-            // Cut off surrounding quotes.
-            workerFile = workerFile.slice(1, -1);
-
-            if (!/^\.{1,2}\//.test(workerFile)) {
+                matchIndex
+              );
+              longWarningAlreadyShown = true;
+            } else {
               this.warn(
-                `Paths passed to the Worker constructor must be relative to the current file, i.e. start with ./ or ../ (just like dynamic import!). Ignoring "${workerFile}".`,
+                `rollup-plugin-off-main-thread: Treating \`${fullMatchWithOpts}\` as \`${fullReplacement}\``,
```

Comment:
Why was this warning message changed? It provided more info than the new one.

Reply:
Ah, it seems to be a diff issue matching different lines and the old one is still below?